framework/src/vectorpostprocessors/SamplerBase.C

```diff
@@ -41,6 +44,10 @@ SamplerBase::SamplerBase(const InputParameters & parameters,
   : _sampler_params(parameters),
     _vpp(vpp),
     _comm(comm),
+    _sampler_transient(dynamic_cast<TransientBase *>(
```

Comment:
Can this cast fail?

Reply:
Yes, but it should just return a nullptr, right? That's checked before calling it below.